src/main/groovy/com/github/winplay02/gitcraft/pipeline/CommitStep.java

```diff
@@ -110,7 +111,14 @@ private void clearWorkingTree(RepoWrapper repo) throws IOException {
 	}
 
 	protected String getBranchNameForVersion(OrderedVersion mcVersion) {
-		return (MinecraftVersionGraph.isVersionNonLinearSnapshot(mcVersion) ? mcVersion.launcherFriendlyVersionName() : !mcVersion.hasClientCode() ? GitCraft.config.gitOldServerLinearBranch : GitCraft.config.gitMainlineLinearBranch).replace(" ", "-");
+		OrderedVersion branch = GitCraft.versionGraph.walkToPreviousBranchPoint(mcVersion);
+		OrderedVersion root = GitCraft.versionGraph.walkToRoot(branch);
+		Set<OrderedVersion> roots = GitCraft.versionGraph.getRootVersions();
+		return (branch == null
+			? roots.size() == 1 || root == GitCraft.versionGraph.getDeepestRootVersion()
+				? GitCraft.config.gitMainlineLinearBranch
+				: root.launcherFriendlyVersionName()
+			: branch.launcherFriendlyVersionName()).replace(" ", "-");
```

Comment:
After a bit of practical testing, this can fail, if `branch` is `null`, which will cause `root` to be `null`, which causes an exception in line 120. I propose to change this line `mcVersion.launcherFriendlyVersionName()`, to make it work.

Reply:
`root` being `null` seems like the problem, there should always be a root - should pass `mcVersion` into `walkToRoot` instead then

Reply:
Perfect, this seems to work